src/mon/OSDMonitor.cc

```diff
@@ -7657,7 +7657,7 @@ bool OSDMonitor::prepare_pool_op(MonOpRequestRef op)
     case POOL_OP_CREATE_SNAP:
     case POOL_OP_DELETE_SNAP:
       if (!pool->is_unmanaged_snaps_mode()) {
-        bool snap_exists = pool->snap_exists(m->name.c_str());
+        bool snap_exists = pool->snap_exists(m->name.c_str()) > 0;
```

Comment:
i don't see the problem of the conversion to bool. it's well define in C++ standard 4.12 [conv.bool].